pom.xml

```diff
@@ -245,6 +248,16 @@
                             </exclusion>
                         </exclusions>
                     </dependency>
+                    <dependency>
+                        <groupId>com.fasterxml.jackson.module</groupId>
+                        <artifactId>jackson-module-scala_${scala.compat.version}</artifactId>
+                        <version>${jackson-module.version}</version>
+                    </dependency>
```

Comment:
I've taken a look at this dependency again. The error message that we get when we don't define it explicitly here is this: `Caused by: java.lang.ClassNotFoundException: com.fasterxml.shaded.jackson.module.scala.ScalaObjectMapper` At first, one might wonder why, because the `ScalaObjectMapper` class is available in the `jackson-module-scala_2.12-2.12.3.jar` which is provided in the `$SPARK_HOME/jars` folder. However, looking closely at the exception, we see that it expects `ScalaObjectMapper` to be in the shaded package. That's because we relocate the whole fasterxml package, and have delta-core as a compile-time dependency, so the reference to `ScalaObjectMapper` in delta-core gets rewritten to the shaded path. 

So, indeed there are three possibilities: 1) Explicitly add the fasterxml jackson packages here, or 2) make delta-core a provided dependency, then the reference to fasterxml will obviously not be rewritten during compile time, or 3) remove the fasterxml relocation. I believe this is required only for spark 2

Reply:
Indeed, I've just tried and ran it, you can remove the fasterxml relocation for spark 3, then you don't have to explictly add the dependency here.

Reply:
Done